plugin/pkg/shoot/validator/admission.go

```diff
@@ -310,6 +310,19 @@ func (v *ValidateShoot) Admit(a admission.Attributes, o admission.ObjectInterfac
 		}
 	}
 
+	// Prevent Shoots from getting hibernated in case they have problematic webhooks.
+	// Otherwise, we can never wake up this shoot cluster again.
+	oldHibernated := oldShoot.Spec.Hibernation != nil && oldShoot.Spec.Hibernation.Enabled != nil && *oldShoot.Spec.Hibernation.Enabled == true
+	newHibernated := shoot.Spec.Hibernation != nil && shoot.Spec.Hibernation.Enabled != nil && *shoot.Spec.Hibernation.Enabled == true
+
+	if !oldHibernated && newHibernated {
+		if hibernationConstraint := helper.GetCondition(shoot.Status.Constraints, garden.ShootHibernationPossible); hibernationConstraint != nil {
+			if hibernationConstraint.Status == garden.ConditionFalse {
+				return admission.NewForbidden(a, fmt.Errorf(hibernationConstraint.Message))
+			}
+		}
+	}
```

Comment:
Can we have a test for this part of the admission plugin?

Reply:
Sure 👍